File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter04/Sponsor/MathLibrary/SimpleMath.cs
using System;

namespace MathLibrary
{
   public class SimpleMath : MarshalByRefObject
   {
      public SimpleMath()
      {
         Console.WriteLine("SimpleMath ctor called");
      }

      public int Add(int n1, int n2)
      {
         Console.WriteLine("SimpleMath.Add({0},{1})", n1, n2);
         return n1 + n2;
      }

      public int Subtract(int n1, int n2)
      {
         Console.WriteLine("SimpleMath.Subtract({0},{1})", n1, n2);
         return n1 - n2;
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter04/RemotingConfig/MathLibrary/SimpleMath.vb
Public Class SimpleMath
   Inherits MarshalByRefObject
      
   Public Sub New()
      Console.WriteLine("SimpleMath ctor called")
   End Sub

   Public Function Add(n1 As Integer, n2 As Integer) As Integer
      Console.WriteLine("SimpleMath.Add({0}, {1})", n1, n2)
      Return n1 + n2
   End Function

   Public Function Subtract(n1 As Integer, n2 As Integer) As Integer
      Console.WriteLine("SimpleMath.Subtract({0}, {1})", n1, n2)
      Return n1 - n2
   End Function 
End Class